flexmeasures/api/common/schemas/tests/test_sensor_data_schema.py

```diff
@@ -318,9 +318,9 @@ def test_build_asset_status_data(
     db.session.add(production_price_sensor)
     db.session.flush()
 
-    asset.consumption_price_sensor_id = wind_sensor.id
-    asset.production_price_sensor_id = production_price_sensor.id
-    asset.inflexible_device_sensors = [temperature_sensor]
+    asset.flex_context["consumption-price"] = wind_sensor.id
+    asset.flex_context["production-price"] = [production_price_sensor.id]
+    asset.flex_context["inflexible-sensors"] = [temperature_sensor.id]
```

Comment:
I realize this is work in progress, but I noticed these flex-context fields should follow a slightly different syntax.

```
asset.flex_context["consumption-price"] = {"sensor": wind_sensor.id}
asset.flex_context["production-price"] = {"sensor": production_price_sensor.id}
asset.flex_context["inflexible-device-sensors"] = [temperature_sensor.id]
```
That is, dicts as values in the 1st and 2nd line, and a different key in the 3rd line.


(Semantically, using a wind sensor as a consumption price, and a temperature sensor as an inflexible device, are quite odd choices. But that's off topic, given that those choices stem from a different PR.)

Reply:
thanks for pointing out, i wonder if this will change how we are storing in the DB

Reply:
@nhoening what do you think, we are currently saving to DB like below

```
{
                "consumption-price": 1,
                "production-price": 4,
                "inflexible-device-sensors": [7],
}
```
should it be then

```
{
                "consumption-price": {"sensor": 1},
                "production-price":  {"sensor": 4},
                "inflexible-device-sensors": [7],
}
```

Reply:
In short: Yes.

So it seems the first implementation from earlier this year only allowed to set sensors for all of this. Now we have formulated a policy to allow different formats (see below), and using JSON allows us to implement this flexibility in encoding.

"inflexible-device-sensors" is in name doing exactly that, so that should stay, indeed.

Anything denoting a "price" or "quantity" we recently treat as being given by a number, a time series or a sensor. This is a new policy and we should pull through with it in all places.

Basically, everything which in the code is a `VariableQuantityField` should be encoded like that.

[Here](https://github.com/FlexMeasures/flexmeasures/pull/1144/files#diff-284640f7db72653f4892af69baf6cc54558e1edcbfeacb2b741f86990ca86d6dR41) is the list of new fields that are entering the flex-context soon and most of them are of that kind.

Reply:
Worth taking a look at the documentation section corresponding to the `VariableQuantityField`, which is [here](https://flexmeasures.readthedocs.io/en/v0.23.2/api/notation.html#variable-quantities). These fields allow `str`, `list[dict]` and `dict` values.

Reply:
Thanks @nhoening @Flix6x will look more into the `VariableQuantityField`